persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickDbProvider.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.jdbc
+
+import javax.naming.InitialContext
+import javax.sql.DataSource
+
+import com.typesafe.config.Config
+import slick.jdbc.JdbcBackend
+import slick.jdbc.JdbcBackend.Database
+import slick.util.AsyncExecutor
+
+object SlickDbProvider {
+
+  def apply(dataSource: DataSource, config: Config): JdbcBackend.DatabaseDef = {
+
+    val asyncExecConfig = new AsyncExecutorConfig(config)
+
+    Database.forDataSource(
+      ds = dataSource,
+      maxConnections = Option(asyncExecConfig.maxConnections),
+      executor = AsyncExecutor(
+        name = "AsyncExecutor.default",
+        minThreads = asyncExecConfig.minConnections,
+        maxThreads = asyncExecConfig.numThreads,
+        queueSize = asyncExecConfig.queueSize,
+        maxConnections = asyncExecConfig.maxConnections
+      )
+    )
+  }
+}
+
+class AsyncExecutorConfig(config: Config) {
```

Comment:
From the conversation in https://github.com/lagom/lagom/pull/1090 I had understood that we would try to consolidate on the style of config classes used in `KafkaConfig`.

Reply:
You mean, having a trait, an object and a impl class?

I like it. However, this `AsyncExecutorConfig` should disappear soon. I want to move it to the `akka-persistence-jdbc` plugin. It should live next to the `SlickConfiguration`. There we use another style. Nothing holds us to do suggest the same style there as well.

BTW, good point about making it private. Especially because I want to remove it soon.